content/posts/pandoc/pandoc.md

```diff
@@ -0,0 +1,334 @@
+
+#### Tricks and Tips
+
+**Image Scalling**
+
+The width and height attributes on images are treated specially. When used without a unit, the unit is assumed to be pixels. However, any of the following unit identifiers can be used: px, cm, mm, in, inch and %. There must not be any spaces between the number and the unit. For example:
+
+```markdown
+![](file.jpg){ width=50% }
+```
+Equal to LaTeX
+
+```latex
+\includegraphics[width=0.5\textwidth,height=\textheight]{file.jpg}
+```
+
+When no width or height attributes are specified, the fallback is to look at the image resolution and the dpi metadata embedded in the image file.
+
+**Slides with Columns in Pandoc**
+Current versions of pandoc (i.e., pandoc 2.0 and later) supports [fenced divs](https://pandoc.org/MANUAL.html#fenced-divs). Specially named divs are transformed into columns when targeting a slides format:
+
+```markdown
+# This slide has columns
+
+::: columns
+
+:::: column
+left
+::::
+
+:::: column
+right
+::::
+
+:::
+```
+
+But I found that It is quite buggy and you dont really control the width of each column easily. Thus We can use both Markdown and LaTeX code inside our main file, so that to create a slide that has columns, I recommend to use raw LaTeX code as the following code below:
+
+```latex
+\Begin{columns}
+
+\Begin{column}{0.5\textwidth}
+- Some content here
+\End{column}
+
+\Begin{column}{0.5\textwidth}
+![](image.png)
+\End{column}
+
+\End{columns}
+```
+
+But you need to add this code to your YAML file ( I will explain what is this code in the next section)
+```YAML
+header-includes:
+- \newcommand{\hideFromPandoc}[1]{#1}
+- \hideFromPandoc{
+    \let\Begin\begin
+    \let\End\end
+  }
+```
+
+**Magical things** is that you can also use Markdown inside a Raw LaTeX code \m/
+
+And this is the result: 
+
+![](/posts/pandoc/images/column.png)
+
+#### Preamble
+
+Pandoc lets user to include a custom preamble in the generated output. 
+
+If you just have to use some packages in LaTeX ( not so much ), you can include it in the YAML header inside the `header-includes:` tag. For example I want to use package `multicol` in LaTeX, my header should look like:
+
+```YAML
+header-includes:
+- \usepackage{multicol}
+```
+
+These lines will be included before the document begin in LaTeX file.\
+
+But Preamble in Pandoc give you more features than just to import some packages. We can also create our own PDF template, for presentation, we can re-config the Beamer theme. For example, as the previous example above, I used Ilmenau theme for my slide. By default, It does not have slide numbers ( frame number ). So our goal now is to re-config the theme without changing the theme source code. If you are using LaTeX then it quite easy to deal with that. Either does Pandoc, instead of using `header-includes` tag, we can create a new file called `preamble.tex` - you can put all the configuration in LaTeX to this file the same way you use LaTeX to create a beamer presentation. \
```

Comment:
- ~~But~~ Preamble in Pandoc *gives* ..
- template*. For presentation*, we can ...
- For example~~, asthe previous example above,~~ I used ... for my slide previously. 
- The last 2 sentences are ambiguous, I couldnt really understand 🤔